tensorflow/stream_executor/rocm/rocm_dnn.cc

```diff
@@ -2645,28 +2684,25 @@ port::Status MIOpenSupport::DoPrepareForConvolution(
 
   auto miopen = miopen_->GetHandle(parent_, stream);
 
-  absl::optional<dnn::AlgorithmDesc> algo_desc = algorithm_config.algorithm();
+  absl::optional<dnn::AlgorithmDesc> input_algo_desc =
+      algorithm_config.algorithm();
   size_t scratch_memory_size;
 
-  if (!algo_desc.has_value()) {
+  miopenStatus_t status = miopenStatusSuccess;
+
+  if (!input_algo_desc.has_value()) {
     // With the default algorithm, use MIOpen's heuristics.
     assert(scratch_allocator);
 
+    assert(kind != dnn::ConvolutionKind::FORWARD);
+
     DeviceMemory<uint8> scratch_memory_temp;
     MIOpenAllocatorContext mac(scratch_allocator, stream);
     wrap::miopenSetAllocator(miopen.handle(), MIOpenAllocatorCallback,
                              MIOpenDeallocatorCallback, &mac);
     size_t size_in_bytes;
-    miopenStatus_t status = miopenStatusSuccess;
 
     switch (kind) {
-      case dnn::ConvolutionKind::FORWARD: {
-        status = wrap::miopenConvolutionForwardGetWorkSpaceSize(
-            miopen.handle(), /*filterDesc=*/filter.handle(),
-            /*srcDesc=*/input_nd.handle(), /*convDesc=*/conv.handle(),
-            /*destDesc=*/output_nd.handle(), /*sizeInBytes=*/&size_in_bytes);
-        break;
-      }
       case dnn::ConvolutionKind::BACKWARD_DATA: {
```

Comment:
Out of curiosity, are BACKWARD_DATA and BACKWARD_FILTER not implemented yet?

Reply:
think they are implemented...just want to make sure that I am going down the right path, by getting the forward pass working, and leaving everything else the same. But now thinking of implementing the rest too, as we need to wait for some MIOpen changes to test things

Reply:
this part is now coded up :)